### PR TITLE
k2SSL: A Faster and Better Framework for Self-Supervised Speech Representation Learning

### PR DESCRIPTION
**Pull request by yfyeung**
Paper link: https://arxiv.org/abs/2411.17100v1

### CODE REVIEW

**Comment by yfyeung** [comment]
Part of the resources have been released:

Zipformer Base pre-trained with cross entropy loss: [Checkpoints, logs, and scripts](https://huggingface.co/yfyeung/k2ssl-ssl-librispeech-zipformer-base)
Zipformer Base fine-tuned with pruned RNN-T loss: [Checkpoints, logs, and scripts](https://huggingface.co/yfyeung/k2ssl-asr-librispeech-100h-zipformer-base-pruned-transducer)
Zipformer Base fine-tuned with letter-level CTC loss: [Checkpoints, logs, and scripts](https://huggingface.co/yfyeung/k2ssl-asr-librispeech-100h-zipformer-base-ctc)
Zipformer Base pre-trained manifests including kmeans labels: [Dataset](https://huggingface.co/datasets/yfyeung/librispeech-hubert-base-ls960-iter1-l6-km500)

With these resources, I believe anyone with 8 V100 32G GPUs can easily reproduce our experiments.

**Comment by ezerhouni** [comment]
@yfyeung Thank you very much for the PR and sharing the model weights. Do you plan to release a paper also ?

**Comment by yfyeung** [comment]
> @yfyeung Thank you very much for the PR and sharing the model weights. Do you plan to release a paper also ?

Maybe, but currently I am not sure whether it is suitable for a technical report or just a normal research paper.

**Comment by ming030890** [comment]
Thanks for the great work!

Do we consider merging the PR and so icefall offically support it?

Thanks

**Comment by yfyeung** [comment]
> Thanks for the great work!
> 
> Do we consider merging the PR and so icefall offically support it?
> 
> Thanks

Yes. I will find some time recently to clean-up the code and upload more checkpoints.
